runtime/gc_structs/ClassLoaderClassesIterator.cpp

```diff
@@ -111,6 +111,18 @@ GC_ClassLoaderClassesIterator::nextClass()
 			if ( (result->classLoader == _classLoader) && (NULL != result->arrayClass) ) {
 				/* this class is defined in the loader, so follow its array classes */
 				_nextClass = result->arrayClass;
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+			} else if ( (result->classLoader == _classLoader) && (NULL != result->nullRestrictedArrayClass) ) {
```

Comment:
It seems that the logic here could leak some array classes. In the original code, if ClassA has 2D arrays, `nextClass()` returns classes in the following sequence: ClassA, 1D-arrayClassA, 2D-ArrayClassA, ClassB, etc.

However if ClassA is a flattenable VT, `nextClass()` in the new code returns classes as: ClassA, 1D-arrayClassA, 2D-arrayClassA, ClassB, etc. When `result` is 2D-arrayClassA, both `else if` checks added here will be false (nullrestrictedArray can only be 1-D). 1D-nullRestricedArrayClassA is leaked.